src/viam/ur/module/ur_arm_state.cpp

```diff
@@ -240,15 +240,18 @@ std::future<void> URArm::state_::enqueue_move_request(size_t current_move_epoch,
     if (move_request_) {
         throw std::runtime_error("an actuation is already in progress");
     }
-    return move_request_.emplace(std::move(samples), std::move(arm_joint_positions_stream)).completion.get_future();
+    return move_request_.emplace(std::move(move_command), std::move(arm_joint_positions_stream)).completion.get_future();
 }
 
 bool URArm::state_::is_moving() const {
     const std::lock_guard lock{mutex_};
-    // If we have a move request, but the samples are gone, it means we
-    // sent them to the arm, so as far as we are concerned, the arm is
-    // moving, though that move might fail later.
-    return (move_request_ && move_request_->samples.empty());
+    if (!move_request_) {
+        return false;
+    }
+    if (const auto* v = std::get_if<std::vector<trajectory_sample_point>>(&move_request_->move_command)) {
+        return v->empty();  // issued -> moving
+    }
+    return false;  // pose_sample still pending
```

Comment:
Fair number of thoughts here:

- Better to std::visit consistently.
- I like the old comment. It explained things more thoroughly.
- The `return false` seems wrong: if the pose_sample is still pending, doesn't that mean we are, in fact, moving? The answer to this question, whatever it may be, should be in the updated comment.